packages/frontend/core/src/components/page-detail-editor.tsx

```diff
@@ -39,6 +44,14 @@ function useRouterHash() {
   return useLocation().hash.substring(1);
 }
 
+let extended = false;
+function useEdgelessTemplate() {
+  if (!extended) {
+    EdgelessTemplatePanel.templates.extend(builtInTemplates as TemplateManager);
+    extended = true;
+  }
+}
```

Comment:
No need to use a hook here since it is not dependent on any react states

Reply:
> No need to use a hook here since it is not dependent on any react states

Not so sure about the code style and structure. Where is the correct place to put this extension call?

Reply:
I think you can put it into `packages/frontend/core/src/bootstrap/register-blocksuite-components.ts`